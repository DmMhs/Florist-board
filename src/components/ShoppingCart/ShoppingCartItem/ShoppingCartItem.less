.cart-item-img {
  display: flex;
  flex-basis: 20%;
  min-width: 100px;
  min-height: 100px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}
