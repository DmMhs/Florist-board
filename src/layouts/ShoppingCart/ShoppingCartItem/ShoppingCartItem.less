.cart-item-img {
  display: flex;
  flex-basis: 20%;
  min-width: 100px;
  min-height: 100px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}
@media screen and (max-width: 478px) {
  .cart-item-img {
    width: 175px;
    height: 175px;
    margin: 0 auto;
    border-radius: 50%;
    box-shadow: 2px 2px 5px #ccc;
  }
}

@media screen and (min-width: 478.1px) and (max-width: 576px) {
  .cart-item-img {
    width: 200px;
    height: 200px;
    margin: 0 auto;
    border-radius: 50%;
    box-shadow: 2px 2px 5px #ccc;
  }
}
