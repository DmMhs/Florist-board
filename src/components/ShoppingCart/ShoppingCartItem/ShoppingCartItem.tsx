import React from 'react';
import { NavLink } from 'react-router-dom';

import './ShoppingCartItem.less';
import { CartItem } from '../../../models/CartItem';

const ShoppingCartItem = (
  props: Partial<CartItem> & { cartItemPrice: number }
) => {
  const {
    remove,
    increaseAmount,
    reduceAmount,
    images,
    title,
    id,
    amount,
    cartItemPrice,
    currency
  } = props;
  return (
    <li className="cart-item">
      <div
        className="cart-item-img"
        style={{
          backgroundImage: `url(${images![0]})`
        }}
      />
      <span className="product-name">
        {title}
        <NavLink to={`/product-details/${id}`}>
          <i className="fas fa-info-circle info" />
        </NavLink>
      </span>
      <span className="amount">
        <i
          className="far fa-minus-square minus"
          onClick={
            reduceAmount as
              | ((event: React.MouseEvent<HTMLElement, MouseEvent>) => void)
              | undefined
          }
        />
        {amount}
        <i
          className="far fa-plus-square plus"
          onClick={
            increaseAmount as
              | ((event: React.MouseEvent<HTMLElement, MouseEvent>) => void)
              | undefined
          }
        />
      </span>
      <span>
        {cartItemPrice}
        {currency}
      </span>
      <span>
        <i
          className="far fa-trash-alt remove"
          onClick={
            remove as
              | ((event: React.MouseEvent<HTMLElement, MouseEvent>) => void)
              | undefined
          }
        />
      </span>
    </li>
  );
};

export default ShoppingCartItem;
