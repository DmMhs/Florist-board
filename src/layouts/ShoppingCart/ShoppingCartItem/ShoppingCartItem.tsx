import React, { useContext } from 'react';
import { NavLink } from 'react-router-dom';

import { CartItem } from '../../../models/CartItem';
import { AppContext } from '../../../AppContext';
import './ShoppingCartItem.less';

const ShoppingCartItem = (
  props: Partial<CartItem> & { cartItemPrice: number }
) => {
  const {
    remove,
    increaseAmount,
    reduceAmount,
    images,
    title,
    title_ua,
    id,
    amount,
    cartItemPrice,
    currency
  } = props;
  const context = useContext(AppContext);
  const productTitle = context.state.lang === 'ua' ? title_ua : title;
  return (
    <li className="cart-item">
      <div
        className="cart-item-img"
        style={{
          backgroundImage: `url(${images![0]})`
        }}
      />
      <span className="product-name">
        {productTitle}
        <NavLink to={`/product-details/${id}`}>
          <i className="fas fa-info-circle info" />
        </NavLink>
      </span>
      <span className="amount">
        <i
          className="far fa-minus-square minus"
          onClick={
            reduceAmount as
              | ((event: React.MouseEvent<HTMLElement, MouseEvent>) => void)
              | undefined
          }
        />
        {amount}
        <i
          className="far fa-plus-square plus"
          onClick={
            increaseAmount as
              | ((event: React.MouseEvent<HTMLElement, MouseEvent>) => void)
              | undefined
          }
        />
      </span>
      <span>
        {cartItemPrice}
        {currency}
      </span>
      <span>
        <i
          className="far fa-trash-alt remove"
          onClick={
            remove as
              | ((event: React.MouseEvent<HTMLElement, MouseEvent>) => void)
              | undefined
          }
        />
      </span>
    </li>
  );
};

export default ShoppingCartItem;
