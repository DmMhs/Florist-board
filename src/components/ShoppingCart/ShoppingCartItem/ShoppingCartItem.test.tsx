import React from 'react';
import { shallow } from 'enzyme';

import ShoppingCartItem from './ShoppingCartItem';

it('Home page matches a snapshot', () => {
  const wrapper = shallow(
    <ShoppingCartItem
      remove={() => {}}
      increaseAmount={() => {}}
      reduceAmount={() => {}}
      images={['', '']}
      title="Sunflower"
      id="qwerty"
      amount={10}
      cartItemPrice={150}
      currency={'usd'}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
