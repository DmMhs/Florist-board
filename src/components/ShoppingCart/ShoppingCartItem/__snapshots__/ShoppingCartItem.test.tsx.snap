// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches a snapshot 1`] = `
<li
  className="cart-item"
>
  <div
    className="cart-item-img"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  />
  <span
    className="product-name"
  >
    Sunflower
    <NavLink
      to="/product-details/qwerty"
    >
      <i
        className="fas fa-info-circle info"
      />
    </NavLink>
  </span>
  <span
    className="amount"
  >
    <i
      className="far fa-minus-square minus"
      onClick={[Function]}
    />
    10
    <i
      className="far fa-plus-square plus"
      onClick={[Function]}
    />
  </span>
  <span>
    150
    usd
  </span>
  <span>
    <i
      className="far fa-trash-alt remove"
      onClick={[Function]}
    />
  </span>
</li>
`;
